### ISSUE TITLE
Surpress analyzer warnings for all methods with specific name.

### ISSUE THREAD

**Issue by DasOhmoff** [dotnet/roslyn#44748]
Hello, in my code, my classes have a function called "Initialize". This function is called via reflection, and should only be allowed to be called via reflection, therefore I set it to private. When I do this I get the following Warning message: `IDE0051: Remove unused private members`. 

How can I tell Visual Studio to suppress this warning for all methods named Initialize?
Maybe through some kind of registry expression (regex) functionality?
```
[assembly: SuppressMessage("CodeQuality", "IDE0051:Remove unused private members",
Justification = "This methods are called via Reflection.", Scope = "member",
Target = ".*Inizialize(.*)")]    // <---
```

**Comment by sharwell** [comment]
There is no built-in way to suppress things like this. In a similar request (all methods with a [particular attribute](https://github.com/dotnet/roslyn/issues/39028)), the conclusion was to use a diagnostic suppressor to automatically suppress the diagnostics getting reported. You can see an example of using a diagnostic suppressor to suppress diagnostics according to a solution-specific pattern that the analyzer did not understand [here](https://github.com/dotnet/machinelearning/blob/master/tools-local/Microsoft.ML.InternalCodeAnalyzer/RelaxTestNamingSuppressor.cs).

**Comment by DasOhmoff** [comment]
Hello, I was trying to make this thing work but I did not succeed yet. How exactly does this all work? Is it enough to just put the suppressor code into my project or do I have to do more than that? I could not find what exactly has to be done so that this code gets executed for my solution in the internet.

**Comment by DasOhmoff** [comment]
Ah ok thanks.
